### DISCUSSION TITLE
Error while using Hono RPC with React Native

### DISCUSSION THREAD

**Discussion by actuallyakshat** [honojs/hono#3642] [Q&A]
![image](https://github.com/user-attachments/assets/2f753f3e-9ced-4dd4-a0b4-9ef938363f44)

![image](https://github.com/user-attachments/assets/cd7c8aa2-c46b-4312-b3d2-362241d161ec)

I have a expo react native application and a hono + bun server as a monorepo with the api apptype exported. When I try to use hono/client, i get an error saying that hono/client couldn't be resolved

![image](https://github.com/user-attachments/assets/e4f0b28f-5058-484b-9ef1-848fc94e123e)

I think I have followed the docs in the right way and I wonder how to get over this issue.
Any help would be appreciated.

**Comment by niikeec** [comment]
@actuallyakshat this should help https://github.com/honojs/hono/issues/1773#issuecomment-2041585319

**Comment by sbkobaidze** [comment]
If you are using expo -- add this in your metro config
`config.resolver.resolveRequest = (context, moduleName, platform) => {
  if (moduleName === "hono/client") {
    return {
      type: "sourceFile",
      filePath: path.resolve(
        monorepoRoot,
        "node_modules/hono/dist/client/index.js"
      ),
    };
  }
  return context.resolveRequest(context, moduleName, platform);
};
`

**Reply by prashsamosa** [reply]
thanks sir
